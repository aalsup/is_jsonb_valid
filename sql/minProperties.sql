-- minProperties validation
-- longer is valid
SELECT is_jsonb_valid('{"minProperties":1}', '{"foo":1,"bar":2}');
-- exact length is valid
SELECT is_jsonb_valid('{"minProperties":1}', '{"foo":1}');
-- too short is invalid
SELECT is_jsonb_valid('{"minProperties":1}', '{}');
-- ignores arrays
SELECT is_jsonb_valid('{"minProperties":1}', '[]');
-- ignores strings
SELECT is_jsonb_valid('{"minProperties":1}', '""');
-- ignores other non-objects
SELECT is_jsonb_valid('{"minProperties":1}', '12');