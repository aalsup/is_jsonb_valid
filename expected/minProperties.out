-- minProperties validation
-- longer is valid
SELECT is_jsonb_valid('{"minProperties":1}', '{"foo":1,"bar":2}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- exact length is valid
SELECT is_jsonb_valid('{"minProperties":1}', '{"foo":1}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- too short is invalid
SELECT is_jsonb_valid('{"minProperties":1}', '{}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- ignores arrays
SELECT is_jsonb_valid('{"minProperties":1}', '[]');
 is_jsonb_valid 
----------------
 f
(1 row)

-- ignores strings
SELECT is_jsonb_valid('{"minProperties":1}', '""');
 is_jsonb_valid 
----------------
 f
(1 row)

-- ignores other non-objects
SELECT is_jsonb_valid('{"minProperties":1}', '12');
 is_jsonb_valid 
----------------
 f
(1 row)
